interpreter/cling/lib/Utils/AST.cpp

```diff
@@ -1501,7 +1501,15 @@ namespace utils {
         // No definition, no lookup result.
         return;
       }
-      S->LookupQualifiedName(R, const_cast<DeclContext*>(primaryWithin));
+      auto res =
+          S->LookupQualifiedName(R, const_cast<DeclContext*>(primaryWithin));
+
+      // If the lookup fails and the context is a namespace, try to lookup in
+      // the namespaces brought in by using directives.
+      if (!res && primaryWithin->isNamespace())
+        for (auto* UD : primaryWithin->using_directives())
+          if (NamespaceDecl* NS = UD->getNominatedNamespace())
+            res = S->LookupQualifiedName(R, NS);
```

Comment:
Sema has an interface `Sema::LookupName` that follows the language rules. Can we use that one? We need to create some scope which describes the lookup position but I guess most of the time we care about the TUScope...

Reply:
`Sema::LookupName` is already used (since the inception of this function) on line 1492 for the `TUScope`.  I suspect we did not know (do we know now?) how to create "some scope which describes the lookup position".

Reply:
I suspect we can push a scope with the namespace we want to look from, but we need to be careful to rebuild the scope nesting going from that namespace towards TU.

Reply:
Not sure if it's the best solution, I realized that clang doesn't look into namespaces unless the `RedeclarationKind` is set to `Sema::NotForRedeclaration`

Reply:
The intent of my comment was to avoid duplication of code that already exists in clang. The new version of that patch seems to address it.